src/server/build-server.ts

```diff
@@ -8,11 +8,32 @@ import jwt from "jsonwebtoken";
 import { prisma } from "../setup-db";
 import { CustomError, errorHandler } from "./error-handler";
 import {
-  AuthRequestBodySchema,
-  RequestBody,
-  UserRequestBodySchema,
+  GetUserRequestBody,
+  PostAuthBodySchema,
+  PostUserBodySchema,
+  PostUserRequestBody,
 } from "./schemas";
 
+function isAuthenticated(request: FastifyRequest) {
```

Comment:
Normalmente funções começadas com `is` retornam booleanas. Acredito que aqui `validateAuthentication`, por exemplo, seja um nome mais apropriado.